diff --git a/dist/index.js b/dist/index.js
index 9fd849550098f37f44ac6f9f64ad174cc821759a..44afa9ccbc42dd81e07ffa794b4be89c61d00c37 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -6,7 +6,9 @@ import { isMockFunction } from '@vitest/spy';
 import { processError } from '@vitest/utils/error';

 const MATCHERS_OBJECT = Symbol.for("matchers-object");
-const JEST_MATCHERS_OBJECT = Symbol.for("$$jest-matchers-object");
+// Patched this symbol for storybook, so that @storybook/test can be used in a jest environment as well.
+// Otherwise, vitest will override global jest matchers, and crash.
+const JEST_MATCHERS_OBJECT = Symbol.for("$$jest-matchers-object-storybook");
 const GLOBAL_EXPECT = Symbol.for("expect-global");

 if (!Object.prototype.hasOwnProperty.call(globalThis, MATCHERS_OBJECT)) {
